src/regex/pcre2.cr

```diff
@@ -196,7 +196,7 @@ module Regex::PCRE2
       selected_range = nil
       exists = false
       @regex.each_capture_group do |number, name_entry|
-        if name_entry[2, group_name.bytesize]? == group_name.to_slice
+        if name_entry[2, group_name.bytesize]? == group_name.to_slice && name_entry[2 + group_name.bytesize].zero?
```

Comment:
Both Crystal `String`s and the strings in the name entry table are null-terminated, so we can do:

```suggestion
        if name_entry[2, group_name.bytesize + 1]? == Slice.new(group_name.to_unsafe, group_name.bytesize + 1)
```